src/meta/server_state.cpp

```diff
@@ -3112,5 +3112,517 @@ void server_state::on_query_manual_compact_status(query_manual_compact_rpc rpc)
+        return;
+    }
+
+    auto &old_partition_config = app->partitions[partition_index];
+    const auto old_max_replica_count = old_partition_config.max_replica_count;
+
+#ifdef INCREASE_MAX_REPLICA_COUNT_ONLY
+    dassert_f(new_max_replica_count >= old_max_replica_count,
+              "new_max_replica_count should never be less than old_max_replica_count: ",
+              "app_name={}, app_id={}, partition_index={}, new_max_replica_count={}, "
+              "old_max_replica_count={}",
+              app_name,
+              app_id,
+              partition_index,
+              new_max_replica_count,
+              old_max_replica_count);
+#endif
+
+    if (new_max_replica_count == old_max_replica_count) {
+        dwarn_f("partition-level max_replica_count has been updated, continue to update the next "
+                "partition: app_name={}, app_id={}, partition_index={}, new_max_replica_count={}",
+                app_name,
+                app_id,
+                partition_index,
+                new_max_replica_count);
+
+        update_next_partition_max_replica_count(*app, partition_index, rpc);
+        return;
+    }
+
+    auto &context = app->helpers->contexts[partition_index];
+    if (context.stage == config_status::pending_remote_sync) {
+        ddebug_f("have to wait until another request which is syncing with remote storage "
+                 "is finished, then process the current request of updating max_replica_count: "
+                 "app_name={}, app_id={}, partition_index={}, new_max_replica_count={}",
+                 app_name,
+                 app_id,
+                 partition_index,
+                 new_max_replica_count);
+
+        tasking::enqueue(LPC_META_STATE_HIGH,
+                         tracker(),
+                         [this, rpc, partition_index]() {
+                             update_partition_max_replica_count(partition_index, rpc);
+                         },
+                         server_state::sStateHash,
+                         std::chrono::milliseconds(100));
+        return;
+    }
+
+    dassert_f(context.stage == config_status::not_pending,
+              "invalid config status while updating max_replica_count: context.stage={}, "
+              "app_name={}, app_id={}, partition_index={}, new_max_replica_count={}",
+              enum_to_string(context.stage),
+              app_name,
+              app_id,
+              partition_index,
+              new_max_replica_count);
+
+    context.stage = config_status::pending_remote_sync;
+    context.pending_sync_request.reset();
+    context.msg = nullptr;
+
+    auto new_partition_config = old_partition_config;
+    new_partition_config.max_replica_count = new_max_replica_count;
+    ++(new_partition_config.ballot);
+    context.pending_sync_task =
+        update_partition_max_replica_count_on_remote(new_partition_config, rpc);
+}
+
+task_ptr server_state::update_partition_max_replica_count_on_remote(
```

Comment:
You better comment on every function to point it run which thread pool, like: https://github.com/XiaoMi/rdsn/blob/v2.3.0/src/meta/meta_bulk_load_service.cpp#L48